storage/client_range_test.go

```diff
@@ -150,3 +155,177 @@ func TestRejectFutureCommand(t *testing.T) {
 		t.Errorf("expected 15, got %v", v)
 	}
 }
+
+// TestTxnPutOutOfOrder tests a case where a put operation of an older
+// timestamp comes after a put operation of a newer timestamp in a txn.
+//
+// The test uses "Writer" and "Reader" to reproduce an out-of-order
+// put. Writer executes a put operation in a txn and commits
+// it. Reader executes two get operations on the same key with higher
+// priority (outside of a txn). An out-of-order put when Writer's txn
```

Comment:
(outside of a txn), which pushes the Writer's intent forward in time, so that the Writer's retry happens in the previous attempt's past.
(then lose the An out-of-order put sentence)